packages/compass-aggregations/src/components/pipeline-results-workspace/index.tsx

```diff
@@ -1,24 +1,149 @@
 import React from 'react';
 import { connect } from 'react-redux';
-import type { ConnectedProps } from 'react-redux';
-import { PipelineResultsList } from './pipeline-results-list';
+import type { Document } from 'mongodb';
+import {
+  IconButton,
+  css,
+  spacing,
+  Icon,
+  Button,
+  Body,
+  uiColors,
+  cx,
+} from '@mongodb-js/compass-components';
 
 import type { RootState } from '../../modules';
+import {
+  fetchNextPage,
+  fetchPrevPage,
+  cancelAggregation,
+} from '../../modules/aggregation';
+import { PipelineResultsList } from './pipeline-results-list';
+
+type PipelineResultsWorkspace = {
+  documents: Document[];
+  page: number;
+  perPage: number;
+  loading: boolean;
+  isPrevDisabled: boolean;
+  isNextDisabled: boolean;
+  error?: string;
+  onPrev: () => void;
+  onNext: () => void;
+  onCancel: () => void;
+};
+
+const containerStyles = css({
+  flexGrow: 1,
+  width: '100%',
+  position: 'relative',
+  overflowY: 'scroll',
+});
+
+const topStyles = css({
+  display: 'flex',
+  justifyContent: 'flex-end',
+  alignItems: 'center',
+  gap: spacing[2],
+});
+
+const centeredContentStyles = css({
+  textAlign: 'center',
+  marginTop: spacing[4],
+});
+
+const errorMessageStyles = css({
+  color: uiColors.red.base,
+});
 
 const PipelineResultsWorkspace: React.FunctionComponent<PipelineResultsWorkspace> =
-  ({ documents }) => {
+  ({
+    documents,
+    page,
+    perPage,
+    loading,
+    isPrevDisabled,
+    isNextDisabled,
+    error,
+    onPrev,
+    onNext,
+    onCancel,
+  }) => {
+    const showingFrom = (page - 1) * perPage;
+    const showingTo = showingFrom + (documents.length || perPage);
     return (
-      <PipelineResultsList
-        documents={documents}
-        data-testid="pipeline-results-workspace"
-      ></PipelineResultsList>
+      <div data-testid="pipeline-results-workspace" className={containerStyles}>
+        <div className={topStyles}>
+          {/* todo: should be replaced with spinners for counts (showingFrom, showingTo) */}
+          {loading ? (
+            <Body>Loading ...</Body>
+          ) : (
+            <Body>
+              Showing {showingFrom + 1} – {showingTo}
+            </Body>
+          )}
+          <div>
+            <IconButton
+              aria-label="Previous page"
+              disabled={isPrevDisabled}
+              onClick={() => onPrev()}
+            >
+              <Icon glyph="ChevronLeft" />
+            </IconButton>
+            <IconButton
+              aria-label="Next page"
+              disabled={isNextDisabled}
+              onClick={() => onNext()}
+            >
+              <Icon glyph="ChevronRight" />
+            </IconButton>
```

Comment:
This section should probably take the error state in the account somehow, otherwise I'm getting both "showing documents" and an error message on the same screen:

<img width="747" alt="image" src="https://user-images.githubusercontent.com/5036933/161226425-48e8f86f-c33e-40f4-a490-a0e3dd8c5f19.png">

Reply:
A little out of topic, is it intentional that the pagination and the view selection breaks in 2 lines? seems like a lot of space thrown away

Reply:
the UX comments, i'll be fixing in my next PR.